### PR TITLE
[CA-3185] [CA-3186] refactoring DateField with new UI and using Luxon

### PR DESCRIPTION
**Pull request by moust**
- Enhanced UI for date field
- Better Intl implementation: fields are ordered according to use locale.
- Use [Luxon](https://www.npmjs.com/package/luxon) for date manipulation and validation

### CODE REVIEW

File: src/components/form/formFieldFactory.js [hunk 1]
```diff
@@ -120,7 +120,7 @@ function customFieldComponent(customField, cfg) {
         const checkboxConfig = cfg.required ? { ...baseConfig, validator: checked } : { ...baseConfig }
         return checkboxField(checkboxConfig);
     } else if (customField.dataType === 'date') {
-        return dateField(baseConfig);
+        return dateField(baseConfig, config);
```
Comment (moust) [inline src/components/form/formFieldFactory.js:123]: Forwards app config to give access to user's language.

File: src/components/form/fields/dateField.jsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            return 'yyyy'
+        case 'literal':
+            return part.value;
+    }
+}).join('')
+
+const datetimeValidator = locale => new Validator({
+    rule: (value) => isValued(value) && value.raw.isValid,
+    hint: 'date',
+    parameters: [dateFormat(locale)]
```
Comment (moust) [inline src/components/form/fields/dateField.jsx:153]: The current translation for `validation.date` message string is `Cette date n'est pas valide. Le format demandé est mm/jj/aaa`.
I added the right Intl date format in `parameters` to be able to change translation with `Cette date n'est pas valide. Le format demandé est {format}` for example.
But with this new implementation I don't see how a user could fill an invalid date format... :)

File: src/components/form/fields/dateField.jsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    )
+
+    const error = typeof validation === 'object' && 'error' in validation ? validation.error : null
+
+    return (
+        <FormGroup
+            inputId={inputId}
+            labelText={label}
+            error={error}
+            showLabel={showLabel}
```
Comment (moust) [inline src/components/form/fields/dateField.jsx:76]: Notice an accessibility issue here.
`<FormGroup>` render an HTML `<label>` with a `for` attribute with `inputId` as value. But neither of the following inputs have this id in their attributs.
According to W3C, a label can refer to only one input, but we have 2 of them...
A valid solution would be to replace le HTML rendered by `<FormGroup>` with a `<fieldset>` and `<legend>`tag instead.

File: src/helpers/useDecounce.js
Comment (axuan25) [inline src/helpers/useDecounce.js:1]: File name is `useDe[c]ounce.js` :) ✍️

**Review by axuan25 — APPROVED** [review]

**Pull request merged**